src/main/java/com/joss/conductor/mobile/Locomotive.java

```diff
@@ -179,6 +176,45 @@ public WebElement waitForElement(String id) {
         return waitForElement(PageUtil.buildBy(configuration, id));
     }
 
+    public WebElement waitForElementByAccessibilityIdentifier(AccessibilityIdentifier id) {
+        try {
+            waitForCondition(ExpectedConditions.not(ExpectedConditions.invisibilityOf(driver.findElementByAccessibilityId(id.getId()))));
+        } catch (Exception e) {
+            Logger.info("WaitForElement: Eat exception thrown waiting for condition");
+        }
+
+        int size = driver.findElementsByAccessibilityId(id.getId()).size();
+
+        if (size == 0) {
+            int attempts = 1;
+            while (attempts <= configuration.getRetries()) {
+                try {
+                    Thread.sleep(1000); // sleep for 1 second.
+                } catch (Exception x) {
+                    Assert.fail("Failed due to an exception during Thread.sleep!");
+                    Logger.error(x);
+                }
+
+                size = driver.findElementsByAccessibilityId(id.getId()).size();
+                if (size > 0) {
+                    break;
+                }
+                attempts++;
+            }
+            if (size == 0) {
+                Assert.fail(String.format("Could not find %s after %d attempts",
+                        driver.findElementByAccessibilityId(id.getId()).toString(),
+                        configuration.getRetries()));
+            }
+        }
+
+        if (size > 1) {
+            Logger.error("WARN: There are more than 1 " + driver.findElementByAccessibilityId(id.getId()).toString() + " 's!");
+        }
+
+        return driver.findElementByAccessibilityId(id.getId());
```

Comment:
Here, if you assign the found element(s) to a variable outside of the while loop (instead of using `size`) you can just return first element in the list instead of having to make another driver call (which are pretty slow).